format irregular map keys

woylie opened the pull request:
I just noticed that `JasonSafeFormatter` won't format irregular map keys. I added a test case for this, but I'm not sure what the result should be in this case. The test causes a `Protocol.UndefinedError`.

related to #73

AndrewDryga commented on test/unit/logger_json_datadog_test.exs:
```diff
@@ -235,14 +235,14 @@ defmodule LoggerJSONDatadogTest do
       |> Jason.decode!()
 
-    line = line + 3
+    # line = line + 3
```
```suggestion
```

AndrewDryga commented:
@woylie thank you, another great catch! Can you please remove than one comment line and I'll merge it

woylie commented:
I only added a failing test case. How would you suggest to format such a map key? Maybe inspect it, so that we have a string key?

AndrewDryga commented:
@woylie my review was too fast, sorry. Almost missed it :/.

I think you are right, the only way to encode it is to use `inspect/1`.

woylie commented:
I updated the PR. I added a property test with StreamData to ensure we don't miss anything (and I would have without it). This should be ready to merge now.

woylie commented:
Just pushed some more minor fixes, and updated the travis config so it doesn't fail.

woylie commented:
The reported coverage dropped after upgrading excoveralls to the latest version.

woylie commented:
I'm done here, should be ready to merge.

AndrewDryga commented:
Thank you @woylie ❤️
